source/base/polynomials_vector_anisotropic.cc

```diff
@@ -0,0 +1,301 @@
+//
+// This file is part of the deal.II library.
+//
+// Part of the source code is dual licensed under Apache-2.0 WITH
+// LLVM-exception OR LGPL-2.1-or-later. Detailed license information
+// governing the source code and code contributions can be found in
+// LICENSE.md and CONTRIBUTING.md at the top level directory of deal.II.
+//
+// ------------------------------------------------------------------------
+
+
+#include <deal.II/base/polynomials_vector_anisotropic.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/thread_management.h>
+
+#include <iomanip>
+#include <iostream>
+#include <memory>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+
+namespace
+{
+  // Create nodal anisotropic polynomials as the tensor product of Lagrange
+  // polynomials on Gauss-Lobatto points of the given degrees in the normal and
+  // tangential directions, respectively (we could also choose Lagrange
+  // polynomials on Gauss points but those are slightly more expensive to handle
+  // in classes).
+  std::vector<std::vector<Polynomials::Polynomial<double>>>
+  create_aniso_polynomials(const unsigned int dim,
+                           const unsigned int normal_degree,
+                           const unsigned int tangential_degree)
+  {
+    std::vector<std::vector<Polynomials::Polynomial<double>>> pols(dim);
+    if (normal_degree > 0)
+      pols[0] = Polynomials::generate_complete_Lagrange_basis(
+        QGaussLobatto<1>(normal_degree + 1).get_points());
+    else
+      pols[0] = Polynomials::generate_complete_Lagrange_basis(
+        QMidpoint<1>().get_points());
+    if (tangential_degree > 0)
+      for (unsigned int d = 1; d < dim; ++d)
+        pols[d] = Polynomials::generate_complete_Lagrange_basis(
+          QGaussLobatto<1>(tangential_degree + 1).get_points());
+    else
+      for (unsigned int d = 1; d < dim; ++d)
+        pols[d] = Polynomials::generate_complete_Lagrange_basis(
+          QMidpoint<1>().get_points());
+
+    return pols;
+  }
+} // namespace
+
+
+
+template <int dim>
+PolynomialsVectorAnisotropic<dim>::PolynomialsVectorAnisotropic(
+  const unsigned int               normal_degree,
+  const unsigned int               tangential_degree,
+  const std::vector<unsigned int> &polynomial_ordering)
+  : TensorPolynomialsBase<dim>(std::min(normal_degree, tangential_degree),
+                               n_polynomials(normal_degree, tangential_degree))
+  , normal_degree(normal_degree)
+  , tangential_degree(tangential_degree)
+  , polynomial_space(
+      create_aniso_polynomials(dim, normal_degree, tangential_degree))
+  , lexicographic_to_hierarchic(polynomial_ordering)
+{
+  // create renumbering of the unknowns from the lexicographic order to the
+  // actual order required by the finite element class with unknowns on
+  // faces placed first
+  const unsigned int n_pols = polynomial_space.n();
+
+  hierarchic_to_lexicographic =
+    Utilities::invert_permutation(lexicographic_to_hierarchic);
```

Comment:
I would move this away from the body of the construction up to the initialization list (after line 73), where you can write `hierarchic_to_lexicographic(Utilities::invert_permutation(lexicographic_to_hierarchic))`